pkg/controllers/controlplanemachineset/watch_filters_test.go

```diff
@@ -91,6 +79,52 @@ var _ = Describe("Watch Filters", func() {
 		}
 	}
 
+	Context("filterClusterOperator", func() {
+		const operatorName = "control-plane-machine-set"
+
+		var clusterOperatorPredicate predicate.Predicate
+
+		BeforeEach(func() {
+			clusterOperatorPredicate = filterClusterOperator(operatorName)
+		})
+
+		It("Panics with the wrong object kind", func() {
+			expectedMessage := "expected to get an of object of type configv1.ClusterOperator"
+			machine := resourcebuilder.Machine().Build()
+			Expect(func() {
+				clusterOperatorPredicate.Create(createEvent(machine))
+			}).To(PanicWith(expectedMessage), "A programming error occurs when passing the wrong object, the function should panic")
+
+			Expect(func() {
+				clusterOperatorPredicate.Update(updateEvent(machine))
+			}).To(PanicWith(expectedMessage), "A programming error occurs when passing the wrong object, the function should panic")
+
+			Expect(func() {
+				clusterOperatorPredicate.Delete(deleteEvent(machine))
+			}).To(PanicWith(expectedMessage), "A programming error occurs when passing the wrong object, the function should panic")
+
+			Expect(func() {
+				clusterOperatorPredicate.Generic(genericEvent(machine))
+			}).To(PanicWith(expectedMessage), "A programming error occurs when passing the wrong object, the function should panic")
+		})
+
+		It("returns false when the wrong cluster operator is provided", func() {
+			co := resourcebuilder.ClusterOperator().WithName("machine-api-operator").Build()
```

Comment:
Nit: Please include a new line between the declarations and expectations

```suggestion
			co := resourcebuilder.ClusterOperator().WithName("machine-api-operator").Build()
			
```